specification/cognitiveservices/OpenAI.Inference/models/chat.completions.tsp

```diff
@@ -0,0 +1,200 @@
+    logitBias?: Record<int32>;
+
+    @doc("""
+    An identifier for the caller or end user of the operation. This may be used for tracking
+    or rate-limiting purposes.
+    """)
+    @projectedName("json", "user")
+    user?: string;
+
+    @doc("""
+    The number of chat completions choices that should be generated for a chat completions
+    response.
+    Because this setting can generate many completions, it may quickly consume your token quota.
+    Use carefully and ensure reasonable settings for max_tokens and stop.
+    """)
+    @projectedName("json", "n")
+    @projectedName("csharp", "ChoiceCount")
+    n?: int32;
+
+    @doc("""
+    A collection of textual sequences that will end completions generation.
+    """)
+    @projectedName("json", "stop")
+    @projectedName("csharp", "StopSequences")
+    stop?: string[];
+
+    @doc("""
+    A value that influences the probability of generated tokens appearing based on their existing
+    presence in generated text.
+    Positive values will make tokens less likely to appear when they already exist and increase the
+    model's likelihood to output new topics.
+    """)
+    @projectedName("json", "presence_penalty")
+    presencePenalty?: float32;
+
+    @doc("""
+    A value that influences the probability of generated tokens appearing based on their cumulative
+    frequency in generated text.
+    Positive values will make tokens less likely to appear as their frequency increases and
+    decrease the likelihood of the model repeating the same statements verbatim.
+    """)
+    @projectedName("json", "frequency_penalty")
+    frequencyPenalty?: float32;
+
+    @doc("""
+    A value indicating whether chat completions should be streamed for this request.
+    """)
+    @projectedName("json", "stream")
+    @projectedName("csharp", "InternalShouldStreamResponse")
+    stream?: boolean;
+
+    @doc("""
+    The model name to provide as part of this completions request.
+    Not applicable to Azure OpenAI, where deployment information should be included in the Azure
+    resource URI that's connected to.
+    """)
+    @projectedName("json", "model")
+    @projectedName("csharp", "InternalNonAzureModelName")
+    "model"?: string;
+};
+
+@doc("""
+The representation of a single prompt completion as part of an overall chat completions request.
+Generally, `n` choices are generated per provided prompt with a default value of 1.
+Token limits and other settings may limit the number of choices generated.
+""")
+model ChatChoice {
+    @doc("The chat message for a given chat completions prompt.")
+    @projectedName("json", "message")
+    message?: ChatMessage;
+
+    @doc("The ordered index associated with this chat completions choice.")
+    @projectedName("json", "index")
+    index: int32;
+
+    @doc("The reason that this chat completions choice completed its generated.")
+    @projectedName("json", "finish_reason")
+    finishReason: CompletionsFinishReason;
```

Comment:
Since this property can be explicitly `null`, should it be marked as `CompletionsFinishReason | null`?

Reply:
Updated.